components/camel-aws-sqs/src/main/java/org/apache/camel/component/aws/sqs/SqsEndpoint.java

```diff
@@ -333,9 +352,13 @@ AmazonSQS createClient() {
                 clientBuilder = AmazonSQSClientBuilder.standard().withClientConfiguration(clientConfiguration);
             }
         }
-        if (ObjectHelper.isNotEmpty(configuration.getRegion())) {
-            clientBuilder = clientBuilder.withRegion(Regions.valueOf(configuration.getRegion()));
-        }
+
+        String host = configuration.getAmazonAWSHost();
+        String region = configuration.getRegion();
+
+        log.debug("Creating endpoint for host {} on region {}", host, region);
+        clientBuilder.withEndpointConfiguration(new AwsClientBuilder.EndpointConfiguration(host, region));
```

Comment:
For the region you may need to use the Regions.valueOf(configuration.getRegion()) because all the components use the enum version so you would have EU_WEST_1, while the endpointConfiguration is expecting the signing region eu-west-1.
I think it should be ok, otherwise. Did you test this with a real AWS SQS queue? Thanks.

Reply:
Thanks for the review!

I have adjusted the points you raised, but I haven't tested on a real SQS queue. Let me do that and get back to you as soon as I verify that.

Reply:
@oscerd I did find an issue with signing after running some AWS tests, but I updated the PR to resolve the problems I found. The PR now works for both the "normal AWS" use case as well as with the custom AWS Host I proposed to support.